From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 5 Aug 2024 11:04:09 +0200
Subject: demux: mkv: write the actual extra data size in the ALAC extradata

This is what avcodec/audio expects when detecting the extradata.

See InitDecoderConfig()

(cherry picked from commit d2265e981fa19a9f5f1299e968348ed3ca98ee60)
---
 modules/demux/mkv/matroska_segment_parse.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/demux/mkv/matroska_segment_parse.cpp b/modules/demux/mkv/matroska_segment_parse.cpp
index 2b93674..706733e 100644
--- a/modules/demux/mkv/matroska_segment_parse.cpp
+++ b/modules/demux/mkv/matroska_segment_parse.cpp
@@ -60,7 +60,7 @@ static inline void fill_extra_data_alac( mkv_track_t *p_tk )
        https://github.com/macosforge/alac/blob/master/ALACMagicCookieDescription.txt */
     SetDWBE( p_extra, p_tk->fmt.i_extra );
     memcpy( p_extra + 4, "alac", 4 );
-    SetDWBE( p_extra + 8, 0 );
+    SetDWBE( p_extra + 8, p_tk->i_extra_data );
     memcpy( p_extra + 12, p_tk->p_extra_data, p_tk->i_extra_data );
 }
 
